bgpd/bgp_route.c

```diff
@@ -804,8 +809,7 @@ static int bgp_path_info_cmp(struct bgp *bgp, struct bgp_path_info *new,
 		return 1;
 	}
 
-	if (!(exist->sub_type == BGP_ROUTE_NORMAL ||
-	      exist->sub_type == BGP_ROUTE_IMPORTED)) {
+	if (!new_origin && exist_origin) {
```

Comment:
should we not add check for if(new_origin && exist_origin) then return 0. This is to ensure we keep old one and do an early exit here?

Reply:
no because then we should follow the other best path decision points for what route wins right?

Reply:
just to understand if we have two route with any of below type then we simple selecte old existing and exit from there. We do not need to run rest of the best path selection as this type of route is most prefered anyways. 

1. BGP_ROUTE_STATIC
 2. BGP_ROUTE_AGGREGATE
 3. BGP_ROUTE_REDISTRIBUTE

Reply:
I disagree.  If you use the old then you have no way to create a reproducible best path as that whomever happens to get in first wins( which is what we are changing )